packages/cli/src/lib/git.ts

```diff
@@ -66,3 +67,107 @@ async function findAndSanitizeRepository(dir: string): Promise<SanitizedUri | vo
 export async function findRepository(dir: string): Promise<string | undefined> {
   return (await findAndSanitizeRepository(dir))?.toString();
 }
+
+/**
+ * Find the distance between the base commit and the head commit.
+ *
+ * @param {string} [base='HEAD'] - The base commit to compare against.
+ * @param {string} [head='HEAD'] - The head commit to compare against.
+ * @param {string} [cwd] - The working directory to run the command in.
+ * @returns {Promise<number | undefined>} A promise containing the distance between the base and
+ * head commits, or undefined if the distance could not be resolved.
+ */
+async function getCommitDistance(
+  base: string,
+  head: string,
+  cwd?: string
+): Promise<number | undefined> {
+  try {
+    const result = await execute('git', ['rev-list', '--count', `${base}..${head}`], { cwd });
+    const distance = parseInt(result);
+    return Number.isNaN(distance) ? undefined : distance;
+  } catch {
```

Comment:
We should probably warn here and another exceptions that aren’t logged.

Reply:
I've added verbose logging for these. In this case, the exception is expected as `git` will fail on branches that don't exist. We're iterating through main, master, develop, so it's likely to occur.